Record exception in Celery instrumentation

tombruijn opened the pull request:
# Description

## Add tests for errors in Celery tasks
I noticed there were no tests for the error scenario in the Celery
package. This commit adds a basic test, based on the previous test and
how I see other packages test the error status on the span.

Part of https://github.com/open-telemetry/opentelemetry-python-contrib/issues/987

## Record exception in Celery instrumentation

In addition to setting the status on the span, also record the exception
on the span. This adds an event to the span with more details about the
error, following the format other instrumentations also use.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] `tox -e test-instrumentation-celery`

# Does This PR Require a Core Repo Change?

- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added

shalevr commented:
Looking great!

shalevr commented:
You can run just the python code formatter, it will fix most of the problems
"black --config pyproject.toml . "

tombruijn commented:
Thanks! That's a lot quicker than running `tox -e lint`.

Does this PR also need an update to the docker-tests? Like you mentioned in the Redis query sanitization PR? https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1572#issuecomment-1383083900

shalevr commented:
> Thanks! That's a lot quicker than running `tox -e lint`.
> 
> Does this PR also need an update to the docker-tests? Like you mentioned in the Redis query sanitization PR? [#1572 (comment)](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1572#issuecomment-1383083900)

There is no need to add docket-tests in this case, In your last PR you changed the default behavior so you had to update all Redis tests(including the docker tests)

tombruijn commented:
Great! Then this PR is ready for review.

srikanthccv commented on CHANGELOG.md:
```diff
@@ -25,6 +27,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#1555](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1555))
 - `opentelemetry-instrumentation-asgi` Fix keys() in class ASGIGetter to correctly fetch values from carrier headers.
   ([#1435](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1435))
+- `opentelemetry/sdk/extension/aws` Implement [`aws.ecs.*`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/cloud_provider/aws/ecs.md) and [`aws.logs.*`](https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/cloud_provider/aws/logs/) resource attributes in the `AwsEcsResourceDetector` detector when t ...
+  ([#1212](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1212))
```
Looks like this is redundant here and mistakenly duplicated from Added section

tombruijn replied:
Looks like that happened with the [latest merge commit](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1573/commits/6c6978e8facab064c2289e105859505030f92d7f).

srikanthccv replied:
Thanks, can you move this to #Added where it originally existed?

tombruijn commented:
@srikanthccv I've updated the PR to fix the failing CI. It's become of some dependency issue described in https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1573/commits/2888c12172024a764e246303d2fdc8bdb28529e4.

I've moved the error test assertion to the functional tests so the build is green.

Can you run the build again?

tombruijn commented:
@srikanthccv I'm learning a lot about what checks need to pass. (I can't run the tox lint task in its entirety locally (some C extension error), so it's process.) The flake8 output contains many issues that do not apply to this PR. I fixed what was related to the celery instrumentation. I hope that's enough. Please try the build again. I really hope it will now be green.

srikanthccv commented:
If you think we can improve/document the steps based on your experience, please suggest them. We want our contributing experience to be better. Whenever I try to reproduce some of these things, I fail because sometimes the errors occur in specific env.

tombruijn commented:
@srikanthccv I asked for help on Slack and Diego was able to help out, resulting in PR #1604. I may send in a PR or two for other things I ran into.
